src/components/explorer/ExplorerSearchResultsTable.js

```diff
@@ -83,8 +99,8 @@ const ExplorerSearchResultsTable = ({
                 key: "all-data",
                 text: "Select All Data",
                 onSelect: () => {
-                    const allRowKeys = data.map((item) => item.key);
-                    handleSetSelectedRows(allRowKeys);
+                    const filteredKeys = filteredData.map((item) => item.key);
```

Comment:
memo dependency should be on filteredData now, not data

Reply:
good catch! Thanks